### PR TITLE
Kata afternoon

### PR DESCRIPTION
**Pull request by lemonses**

### CODE REVIEW

File: __tests__/server.test.js [hunk 1]
```diff
@@ -124,6 +125,32 @@ describe('GET /api/articles',()=>{
...
+        return request(app)
+        .get('/api/articles?topic=mitch')
+        .expect(200)
+        .then(({body})=>{
+            body.articles.forEach((article)=>{
+                expect(article.topic).toBe('mitch')
+            })
+        })
+    })
+    test('should return a 200 with an empty array if given a valid topic with no articles',()=>{
```
Comment (RichardNeat) [inline __tests__/server.test.js:144]: Really good consideration here!

File: server/models/articles.model.js [hunk 2]
```diff
@@ -12,8 +20,17 @@ exports.fetchArticle = (article_id) => {
...
 }
 
-exports.fetchArticles = () => {
-    return db.query(`
+exports.fetchArticles = (topic = '%') => {
+    const topicQuery = format(`
+    SELECT * FROM topics
+    WHERE slug LIKE %L;
+    `,topic)
+    return db.query(topicQuery) //refactor this to be its own function?
```
Comment (RichardNeat) [inline server/models/articles.model.js]: Please remove comments before committing. Although I agree this being a utils function that we call alongside this model in the controller with a promise.all could be a nice way of handling it 👍

File: __tests__/server.test.js [hunk 3]
```diff
@@ -124,6 +125,32 @@ describe('GET /api/articles',()=>{
...
+    })
+    test('should return a 404 Topic not found if the topic does not exist',()=>{
+        return request(app)
+        .get('/api/articles?topic=notATopic')
+        .expect(404)
+        .then(({body})=>{
+            expect(body.message).toBe('Topic not found')
+        })
+    })
 })
```
Comment (RichardNeat) [inline __tests__/server.test.js:160]: Do we have a test anywhere for the new comment_count property on a single article?

**Review by RichardNeat — CHANGES_REQUESTED** [review]
Nice one, just a couple of requested changes below please. The main one being for task 12 we currently don't have any testing for the new comment_count property on getArticleById

File: __tests__/server.test.js (hunk 3 above)
Comment (lemonses) [reply __tests__/server.test.js:160]: I only updated the previous test I didn't include a separate test but I will add one

Comment (RichardNeat) [reply __tests__/server.test.js:160]: Gotcha - we should never update previous tests. This is why we don't use brittle jest matchers like toEqual to allow for future expansion of our endpoints. We should just have one new test to check for this one new property on this endpoint and leave the previous test as it was. If done correctly both should pass.

**Pull request merged**